pkg/tf2pulumi/convert/eject_test.go

```diff
@@ -263,51 +264,43 @@ func TestEject(t *testing.T) {
 			assert.Equal(t, tokens.PackageName(tt.name), project.Name)
 
 			// Assert every pcl file is seen
-			infos, err := os.ReadDir(pclPath)
+			_, err = os.ReadDir(pclPath)
 			if !os.IsNotExist(err) && !assert.NoError(t, err) {
 				// If the directory was not found then the expected pcl results are the empty set, but if the
 				// directory could not be read because of filesystem issues than just error out.
 				assert.FailNow(t, "Could not read expected pcl results")
 			}
-			pclFiles := make(map[string]interface{})
-			// infos will just be nil if pclPath did not exist
-			for _, info := range infos {
-				if !info.IsDir() {
-					pclFiles[info.Name()] = nil
-				}
-			}
+
+			pclFs := afero.NewBasePathFs(afero.NewOsFs(), pclPath)
+			pclMemFs := afero.NewMemMapFs()
 
 			// If PULUMI_ACCEPT is set then clear the PCL folder and write the generated files out
 			if isTruthy(os.Getenv("PULUMI_ACCEPT")) {
-				err := os.RemoveAll(pclPath)
-				require.NoError(t, err)
-				err = os.Mkdir(pclPath, 0700)
-				require.NoError(t, err)
-				for filename, source := range program.Source() {
-					// normalize windows newlines to unix ones
-					expectedPcl := []byte(strings.Replace(source, "\r\n", "\n", -1))
-					err := os.WriteFile(filepath.Join(pclPath, filename), expectedPcl, 0600)
-					require.NoError(t, err)
-				}
+				err := pclFs.RemoveAll(pclPath)
+				require.NoError(t, err, "failed to remove existing files at %s", pclPath)
+				err = program.WriteSource(pclFs)
+				require.NoError(t, err, "failed to write program source files")
 			}
 
-			// Assert the pcl source is as expected
-			for filename, source := range program.Source() {
-				pclBytes, err := os.ReadFile(filepath.Join(pclPath, filename))
-				if assert.NoError(t, err) {
-					// normalize windows newlines
-					expectedPcl := strings.Replace(string(pclBytes), "\r\n", "\n", -1)
-					source = strings.Replace(source, "\r\n", "\n", -1)
-					assert.Equal(t, expectedPcl, source)
-					delete(pclFiles, filename)
+			// Write the program to a memory file system
+			program.WriteSource(pclMemFs)
+
+			// compare the two
+			afero.Walk(pclMemFs, "/", func(path string, info fs.FileInfo, err error) error {
+				if info == nil || info.IsDir() {
+					// ignore directories
+					return nil
 				}
-			}
 
-			unseenPcl := make([]string, 0)
-			for name := range pclFiles {
-				unseenPcl = append(unseenPcl, name)
-			}
-			assert.Empty(t, unseenPcl)
+				sourceOnDisk, err := afero.ReadFile(pclFs, path)
+				assert.NoError(t, err, "Must have seen this file before")
```

Comment:
Because we are traversing files in `pclMemFs`, so each file written to memory is expected to be on disk. The `path` in `afero.ReadFile(pclFs, path)` is from the memory file system and I am expecting it to exist in the real file system on the os. 

Maybe the error message is wrong but I think this way we ensure that every PCL file generated by `WriteSource(fs)` also exists on disk

Reply:
Fixed the error message